html2/tokenizer.cpp

```diff
@@ -214,6 +236,507 @@ void Tokenizer::run() {
+                };
+
+                if (!c) {
+                    anything_else();
+                    continue;
+                }
+
+                if (is_ascii_upper_alpha(*c)) {
+                    auto lower_case_c = to_lower(*c);
+                    std::get<EndTagToken>(current_token_).tag_name.append(1, lower_case_c);
+                    temporary_buffer_.append(1, lower_case_c);
+                    continue;
+                }
+
+                if (is_ascii_lower_alpha(*c)) {
+                    std::get<EndTagToken>(current_token_).tag_name += *c;
+                    temporary_buffer_.append(1, *c);
+                    continue;
+                }
+
+                switch (*c) {
+                    case '\t':
+                    case '\n':
+                    case '\f':
+                    case ' ':
+                        if (is_appropriate_end_tag_token(current_token_)) {
+                            state_ = State::BeforeAttributeName;
+                        } else {
+                            anything_else();
+                        }
+                        continue;
+                    case '/':
+                        if (is_appropriate_end_tag_token(current_token_)) {
+                            state_ = State::SelfClosingStartTag;
+                        } else {
+                            anything_else();
+                        }
+                        continue;
+                    case '>':
+                        if (is_appropriate_end_tag_token(current_token_)) {
+                            state_ = State::Data;
+                            emit(std::move(current_token_));
+                        } else {
+                            anything_else();
+                        }
+                        continue;
+                    default:
+                        anything_else();
+                        continue;
+                }
+            }
+
+            case State::ScriptDataDoubleEscapeStart: {
+                auto c = consume_next_input_character();
+                if (!c) {
+                    reconsume_in(State::ScriptDataEscaped);
+                    continue;
+                }
+
+                if (is_ascii_upper_alpha(*c)) {
+                    auto lower_case_c = to_lower(*c);
+                    temporary_buffer_.append(1, lower_case_c);
+                    emit(CharacterToken{*c});
+                    continue;
+                }
+
+                if (is_ascii_lower_alpha(*c)) {
+                    temporary_buffer_.append(1, *c);
+                    emit(CharacterToken{*c});
+                    continue;
+                }
+
+                switch (*c) {
+                    case '\t':
+                    case '\n':
+                    case '\f':
+                    case ' ':
+                    case '/':
+                    case '>':
+                        if (temporary_buffer_ == "script"s) {
```

Comment:
When comparing you can compare to a `""sv` instead. No need to require an heap-allocated string for this.